doc: Remove Getting Started Assistant documentation (#6909)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

Documentation
=============

* Removed:

  * Documentation on the Getting Started Assistant, as this tool is no longer in use.

>     Linux users can use the `Installing using Visual Studio Code <Installing on Linux_>`_ instructions or follow the steps on the :ref:`gs_installing` page for |NCS| installation.

Reviewer: ```suggestion
    Linux users can install the |NCS| by using the `Installing using Visual Studio Code <Installing on Linux_>`_ instructions or by following the steps on the :ref:`gs_installing` page.
```

  * :ref:`nrfxlib:nrf_modem` - Removed overview diagram.
  * ``AT socket`` page removed.
  * ``Extensions`` page removed.

* Removed:

  * Documentation on the Getting Started Assistant, as this tool is no longer in use.

>     Linux users can use the `Installing using Visual Studio Code <Installing on Linux_>`_ or follow the steps on the :ref:`gs_installing` page for |NCS| installation.

Reviewer 2: Same as the previous note.

```suggestion
    Linux users can use the `Installing using Visual Studio Code <Installing on Linux_>`_ instructions or follow the steps on the :ref:`gs_installing` page for |NCS| installation.
```
